pkg/gce-pd-csi-driver/cache.go

```diff
@@ -175,10 +175,18 @@ func setupCaching(devicePath string, req *csi.NodeStageVolumeRequest, nodeId str
 		klog.V(4).Infof("Assuming valid data cache size and mode, resizing cache is not supported")
 	} else {
 		cacheSize := req.GetPublishContext()[common.ContextDataCacheSize]
-		chunkSize, err := fetchChunkSizeKiB(cacheSize)
+		maxChunkSizeStr := strconv.FormatFloat(maxChunkSize, 'g', -1, 64)
+		var chunkSize string
+		cachePvSize, err := fetchPvSize(raidedLocalSsdPath)
 		if err != nil {
-			klog.Errorf("Errored to fetch cache size, verify the data-cache-size is valid: got %v, error: %q", cacheSize, err)
-			return mainDevicePath, err
+			klog.Errorf("Errored while fetching PV size, got %v, falling back to default chunkSize of %v", err, maxChunkSize)
+			chunkSize = maxChunkSizeStr
+		} else {
+			chunkSize, err = fetchChunkSizeKiB(cachePvSize)
```

Comment:
The fetchChunkSizeKiB function itself is assuming that the input passing in (previously cacheSize now cachePvSize) is in GiB unit. But your fetchPvSize is returning KiB size already? Maybe the logic in fetchChunkSizeKiB also needs to change?